docs/migrating-to-new-building-env.md

```diff
@@ -0,0 +1,57 @@
+---
+permalink: /trustzone-sdk-docs/migrating-to-new-building-env.md
+---
+
+## Migration Guide: Moving from `master` to `main` Branch (Post-Oct 2024)
+
+Since the `main` branch (after October 2024) introduces breaking changes 
+to the build environment, if users of the legacy `master` branch want to 
+keep upstream or use a new version of the Rust toolchain, they will need 
+to follow these steps to migrate their TA to the new environment.
+
+### Step 1: Migrate Old Code to the New Environment
+
+To migrate an example project (e.g., `tls_server-rs` in `examples/`), you 
```

Comment:
IMHO, we should rather be describing the build environment for the `main` branch as to how it's structured. This will help developers writing new TAs or updating existing TAs to the build environment on `main` branch.

A few words for each file under following directory structure should be sufficient:

```
examples/acipher-rs/
├── host
│   ├── Cargo.toml
│   ├── Makefile
│   └── src
│       └── main.rs
├── Makefile
├── proto
│   ├── build.rs
│   ├── Cargo.toml
│   └── src
│       └── lib.rs
├── ta
│   ├── build.rs
│   ├── Cargo.toml
│   ├── Makefile
│   ├── src
│   │   └── main.rs
│   ├── ta_static.rs
│   └── Xargo.toml
└── uuid.txt

```

Reply:
Doc updated, thanks for all comments